service/storage_service.cc

```diff
@@ -1307,6 +1311,13 @@ future<> storage_service::do_update_system_peers_table(gms::inet_address endpoin
         co_await update_table(endpoint, "schema_version", utils::UUID(value.value));
     } else if (state == application_state::HOST_ID) {
         co_await update_table(endpoint, "host_id", utils::UUID(value.value));
+    } else if (state == application_state::RAFT_SERVER_ID && _feature_service.supports_raft_cluster_mgmt) {
```

Comment:
> So we will only save raft_server_id to system.peers when this feature is enabled. I understand that this is necessary because we must not use a column in a local table until every node has this column.
> 

Why? What could happen if we will use it earlier?

Reply:
I believe we concluded that the latest version of this check is correct.